### PR TITLE
Added basic BTS c3d and parsing/rebuiding tests 

### PR DESCRIPTION
**Pull request by marnunez**

### CODE REVIEW

**Comment by marnunez** [comment]
This is strange. Using C++ tests, the Qualisys rebuilt c3d keeps its platforms perfectly. Could this have anything to do with the Python bindings then?

Moreover, BTS c3d writing fails, resulting rebuilt BTS_after.c3d cant be parsed again. Curiously, in Mokka:

Original:
image
Rebuilt:
image

It looks like NaNs/masked values are sent to origin. Oh, and platforms are kept intact. Weird

**Comment by pariterre** [comment]
> Could this have anything to do with the Python bindings then?

Oh, that's it, the parsing in python was a bit harder for the hypermatrix (such as CORNER which has a shape of 3x4xnPF). I probably messed up there, because at the time I made the binder, I did not have C3D to test this. That could explain why I could not reproduce your bug (since I test everything in C++)!

> t looks like NaNs/masked values are sent to origin

I did not do anything specific for the NaNs, I'll have a look on this

**Comment by marnunez** [comment]
It might be useful then to run the python tests at build time.

**Comment by pariterre** [comment]
Yep, I am on it
Actually, some tests are done, but obviously not enough..

**Comment by marnunez** [comment]
Would you like me to work in the python testing part? I'm expanding test_binder_python.py to add similar parsing/building tests python side.

**Comment by marnunez** [comment]
Ok, I think that's it. This are all tests that should pass, the problem affects all parameter blocks and the Qualisys header block. How would you like to proceed?

**Comment by pariterre** [comment]
I am currently changing some stuff in your PR (mostly for the header and some specific things in the parameters) that prevents tests to pass. I will PR into your repo when it's ready. So it can update your PR too and finalize the BTS integration!
Thanks for the work :)

File: binding/python3/__init__.py [hunk 1]
```diff
@@ -33,9 +38,6 @@ def __setitem__(self, key, value):
     def __delitem__(self, key):
         del self._storage[key]
 
-    def __iter__(self):
```
Comment (pariterre) [inline binding/python3/__init__.py:36]: Is there any reason for removing this?

File: binding/python3/__init__.py [hunk 2]
```diff
@@ -21,6 +21,11 @@ def __len__(self):
     def keys(self):
         return self._storage.keys()
 
+    # Am I right to add this? Useful for testing
```
Comment (pariterre) [inline binding/python3/__init__.py]: Yep good idea.
Leave the comment over I will remove it in my own PR to you

File: binding/python3/__init__.py (hunk 1 above)
Comment (marnunez) [reply binding/python3/__init__.py:36]: I believe it was exactly the same in the parent class. I'll check it as soon as I get home

**Comment by marnunez** [comment]
> I did not do anything specific for the NaNs, I'll have a look on this

Now that I think about it, the failing C++ parsing/building test might be related to the NaNs. I'll try a BTS c3d with no point gaps.

**Comment by pariterre** [comment]
No worries, the up coming PR doesn't fail for the BTS. The NaNs problem is probably related to the Python interface

**Comment by pariterre** [comment]
@marnunez 
I PRed some changes to your branch, please accept the PR, that should trigger the tests here!
I think all the problem you pointed should be fixed now :)
Thanks!

File: binding/python3/CMakeLists.txt [hunk 3]
```diff
@@ -62,5 +62,5 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/_version.py.i ${CMAKE_CURRENT_BINARY_
 INSTALL(TARGETS _${PROJECT_NAME}
     DESTINATION ${PYTHON_INSTALL_DESTINATION}
 )
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/_version.py DESTINATION ${PYTHON_INSTALL_DESTINATION}/${PROJECT_NAME})
```
Comment (marnunez) [inline binding/python3/CMakeLists.txt:65]: Actually, I think this was right, I had to manually move _version.py from site-packages to site-packages/ezc3d, both in Windows and Debian, miniconda3, SWIG 3 and 4 both.

**Comment by pariterre** [comment]
@marnunez 
I accept this PR even though the tests didn't pass, because I solved them elsewhere. 
Thanks a lot for the tests! The next PR with the tests that passes, should solve all the issues listed in the that you open (and I have one last very weird problem to fix on Mokka where everything is rotated 90° while creating the c3d from Python or Matlab)

**Comment by marnunez** [comment]
> @marnunez
> I accept this PR even though the tests didn't pass, because I solved them elsewhere.
> Thanks a lot for the tests! The next PR with the tests that passes, should solve all the issues listed in the that you open (and I have one last very weird problem to fix on Mokka where everything is rotated 90° while creating the c3d from Python or Matlab)

If its the BTS c3d, the default ground plane is XZ, not XY. That's ok, it's just representation. Could it be that?

**Comment by pariterre** [comment]
> If its the BTS c3d, the default ground plane is XZ, not XY. That's ok, it's just representation. Could it be that?

It was Mokka that was a bit picky, I was storing X_SCREEN and Y_SCREEN as matrix (2, 1), while Mokka was specifically waiting for a vector (2,). So easy fix when you find the problem :)

I was also storing the marker names trailing spaces ("c7    ") as '\0' as is it usually done in binary, but Mokka was expecting actual space, otherwise it prints the '\0' character as a square. So I also changed that for Mokka!

The dev version should work fine on Mokka as well!

**Pull request merged**